flow/IRandom.h

```diff
@@ -34,6 +34,32 @@
 #endif
 #include <functional>
 
+// Until we move to C++20, we'll need something to take the place of operator<=>.
+// This is as good a place as any, I guess.
+
+template <typename T>
+typename std::enable_if<std::is_integral<T>::value, int>::type compare(T l, T r) {
```

Comment:
The return type should be a signed integer of the same precision, I think something of the form 
`typename std::make_signed<T>::type`
will get you that, and then the implementation can just be `l - r`.

Reply:
I think the above will result in something that optimizes to the same thing as using subtraction directly but with the superior readability of calling `compare()`.

Also, I'm not sure that this file is the best place for these definitions, but then I also don't have a better suggestion so I guess it's fine.